### ISSUE TITLE
Can't inherit a Theme and change the Text Color for a subtree at the same time

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#45232]
I can't find a easy way  to use my current `Theme` on a subtree text, but also be able to change that text color.

For example, lets suppose I have this widget tree:

```dart
Column(  
  children: <Widget>[
    Text(
      'Text 1',
      style: Theme.of(context).textTheme.title,
    ),
    SizedBox(height: 16,),
    Text(
       'Text 2'
       style: Theme.of(context).textTheme.body2,
    ),
  ],
),
```

And I want to change all my texts colors to white, while maintaining their inherit typography styles. Of course I can edit each one of them and change their individual colors to white, but this is tedious, specially for sub-trees with lots of texts.

Using a `DefaultTextStyle` could be a solution, but then I would have to set each individual `Text` `style` as `style: Theme.of(context).textTheme.title.copyWith(color: DefaultTextStyle.of(context).style.color)`, which is A LOT of boilerplate code, and I think that's the opposite of what Flutter stands for.

Another solution is to wrap everything in a `Theme`, then a `DefaultTextStyle`, then a `Builder`, which again, is lots of boilerplate code.

I would expect from Flutter an easy way to accomplish this, a way to just intercept and change the colors from that point forward in the sub-tree, without losing any previously defined configurations, but so far I can't see a simple way to achieve this and I think the Framework should put some thought on how to streamline this.

PS: I firstly introduced this discussion [here](https://stackoverflow.com/questions/57455620/how-to-change-the-text-color-for-a-subtree-in-flutter) and forgot to follow it up in here.

**Comment by iapicca** [comment]
Hi @feinstein 
I believe you did the right thing to open the stackoverflow question,
since this platform is not met for assistance on personal code.
Nevertheless I feel that you are also aiming to improve the experience with `Theme`
do you have a specific proposal you want to forward?
thank you

**Comment by feinstein** [comment]
@iapicca yes, I opened this issue because I think there's room for improvement, but although I know how I wish it worked, I can't provide a solution since I am not familiar with all the inner workings of theming, so my proposal might not make sense at all.

As I said, I wished we could develop a simple way to inherit the main `Theme` but still be able to override some specific settings for the subtree. Do you have an idea on how this can be achieved, specially without the need to create a new `BuildContext`, as to keep it simple?

**Comment by iapicca** [comment]
I'm gonna keep this issue open as proposal,
thank you for your contribution @feinstein
